### PR TITLE
Add Callback documentation

### PR DESCRIPTION
**Pull request by tgallice**
The Callback documentation. My english is not very good, so if you find mistakes or improvements let's me know.

### CODE REVIEW

File: docs/templates/getting_started.md [hunk 1]
```diff
@@ -372,6 +372,54 @@ tflearn.is_training(False)
...
+- `on_train_begin(training_state)`
+- `on_epoch_begin(training_state)`
+- `on_batch_begin(training_state)`
+- `on_sub_batch_begin(training_state)`
+- `on_sub_batch_end(training_state, train_index)`
+- `on_batch_end(training_state, snapshot)`
+- `on_epoch_end(training_state)`
+- `on_train_end(training_state)`
+
+# How to use it:
```
Comment (aymericdamien) [inline docs/templates/getting_started.md]: Could you update it to '####', otherwise title is too large.

**Comment by aymericdamien** [comment]
Great! This was needed. I added a comment, please update it and I will merge it. Thanks! :)

File: docs/templates/getting_started.md (hunk 1 above)
Comment (tgallice) [inline docs/templates/getting_started.md]: :+1: My bad, it's fixed.

**Comment by aymericdamien** [comment]
Great! Thanks!

**Pull request merged**